skrifa/src/outline/glyf/hint/engine/dispatch.rs

```diff
@@ -0,0 +1,181 @@
+//! Instruction decoding and dispatch.
+
+use read_fonts::tables::truetype::bytecode::Opcode;
+
+use super::{Engine, HintError, HintErrorKind, Instruction};
+
+impl<'a> Engine<'a> {
+    /// Decodes and dispatches all instructions until completion or error.
+    ///
+    /// On success, returns the number of instructions executed.
```

Comment:
I immediatley wonder why `Result<(), HintError` doesn't suffice. Perhaps the comment could explain why the # of instructions is useful?

Reply:
I think this is habit from my many debugging sessions. FT prints number of instructions executed at the end of a trace and a mismatch there is usually the first indicator that something is very wrong so I find the info useful. That said, since this now supports “step” debugging with separate decode/dispatch, I can write a nice tracer on that and this is not that useful.